File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRRoom.java
package com.pennapps.labs.pennmobile.classes;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by MikeD on 2/10/2018.
 */

public class GSRRoom {

    @SerializedName("capacity")
    @Expose
    public Integer capacity;

    @SerializedName("gid")
    @Expose
    public Integer gid;


    @SerializedName("lid")
    @Expose
    public Integer lid;


    @SerializedName("name")
    @Expose
    public String name;

    @SerializedName("room_id")
    @Expose
    public Integer room_id;

    @SerializedName("thumbnail")
    @Expose
    public String thumbnail;


    @SerializedName("times")
    @Expose
    public GSRSlot[] slots;

    //getters


    public Integer getCapacity() {
        return capacity;
    }

    public Integer getGid() {
        return gid;
    }

    public Integer getLid() {
        return lid;
    }

    public String getName() {
        return name;
    }

    public Integer getRoom_id() {
        return room_id;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public GSRSlot[] getSlots() {
        return slots;
    }
}

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRRoom.kt
package com.pennapps.labs.pennmobile.classes

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Varun on 10/14/2018.
 */

class GSRRoom {
    // getters

    @SerializedName("capacity")
    @Expose
    var capacity: Int? = null

    @SerializedName("gid")
    @Expose
    var gid: Int? = null

    @SerializedName("lid")
    @Expose
    var lid: String? = null

    @SerializedName("room_name")
    @Expose
    var name: String? = null

    @SerializedName("id")
    @Expose
    var roomId: Int? = null

    @SerializedName("thumbnail")
    @Expose
    var thumbnail: String? = null

    @SerializedName("availability")
    @Expose
    var slots: Array<GSRSlot>? = null
}
